integration_tests/src/lib.rs

```diff
@@ -24,6 +26,324 @@ mod test {
     use tokio::sync::mpsc;
     use uuid::*;
 
+    trait TestDownstairsDataset: Sized {
+        /// Create a new dataset for a region of [`size`] bytes
+        fn new(size: u64) -> Result<Self>;
+
+        /// The path where the Downstairs should run
+        fn path_buf(&self) -> Result<PathBuf>;
+
+        /// Returns true if a snapshot exists
+        fn snapshot_exists(&self, snapshot_name: String) -> Result<bool>;
```

Comment:
This could take a `&str`, which would remove `clone()` calls in `TestDownstairsSet::snapshot_exists`